test/e2e/play_kube_test.go

```diff
@@ -77,14 +88,109 @@ spec:
 status: {}
 `
 
+var deploymentYAMLTemplate = `
+apiVersion: v1
+kind: Deployment
+metadata:
+  creationTimestamp: "2019-07-17T14:44:08Z"
+  labels:
+    app: {{ .Name }}
+  name: {{ .Name }}
+{{ with .Annotations }}
+  annotations:
+  {{ range $key, $value := . }}
+    {{ $key }}: {{ $value }}
+  {{ end }}
+{{ end }}
+
+spec:
+  replicas: {{ .Replicas }}
+  selector:
+    matchLabels:
+      app: {{ .Name }}
+  template:
+  {{ with .PodTemplate }}
+    metadata:
+      labels:
+        app: {{ .Name }}
+      {{ with .Annotations }}
+        annotations:
+        {{ range $key, $value := . }}
+          {{ $key }}: {{ $value }}
+        {{ end }}
+      {{ end }}
+    spec:
+      hostname: {{ .Hostname }}
+      containers:
+    {{ with .Ctrs }}
+      {{ range . }}
+      - command:
+        {{ range .Cmd }}
+        - {{.}}
+        {{ end }}
+        env:
+        - name: PATH
+          value: /usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin
+        - name: TERM
+          value: xterm
+        - name: HOSTNAME
+        - name: container
+          value: podman
+        image: {{ .Image }}
+        name: {{ .Name }}
+        imagePullPolicy: {{ .PullPolicy }}
+        resources: {}
+        {{ if .SecurityContext }}
+        securityContext:
+          allowPrivilegeEscalation: true
+          {{ if .Caps }}
+          capabilities:
+            {{ with .CapAdd }}
+            add:
+              {{ range . }}
+              - {{.}}
+              {{ end }}
+            {{ end }}
+            {{ with .CapDrop }}
+            drop:
+              {{ range . }}
+              - {{.}}
+              {{ end }}
+            {{ end }}
+          {{ end }}
+          privileged: false
+          readOnlyRootFilesystem: false
+        workingDir: /
+        {{ end }}
+      {{ end }}
+    {{ end }}
+  {{ end }}
+`
+
 var (
-	defaultCtrName  = "testCtr"
-	defaultCtrCmd   = []string{"top"}
-	defaultCtrImage = ALPINE
-	defaultPodName  = "testPod"
-	seccompPwdEPERM = []byte(`{"defaultAction":"SCMP_ACT_ALLOW","syscalls":[{"name":"getcwd","action":"SCMP_ACT_ERRNO"}]}`)
+	defaultCtrName        = "testCtr"
+	defaultCtrCmd         = []string{"top"}
+	defaultCtrImage       = ALPINE
+	defaultPodName        = "testPod"
+	defaultDeploymentName = "testDeployment"
+	seccompPwdEPERM       = []byte(`{"defaultAction":"SCMP_ACT_ALLOW","syscalls":[{"name":"getcwd","action":"SCMP_ACT_ERRNO"}]}`)
 )
 
+func writeYAML(content string, fileName string) error {
+	f, err := os.Create(fileName)
+	if err != nil {
+		return err
+	}
+	defer f.Close()
+
+	_, err = f.WriteString(content)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
 func generateKubeYaml(pod *Pod, fileName string) error {
```

Comment:
second nit for this line: we should stick to either YAML or Yaml. I don't care which, but if you'd prefer to use YAML, can you change all of them?